drizzle-arktype/tests/mysql.test.ts

```diff
@@ -0,0 +1,501 @@
+import { Type, type } from 'arktype';
+import { type Equal, sql } from 'drizzle-orm';
+import { customType, int, json, mysqlSchema, mysqlTable, mysqlView, serial, text } from 'drizzle-orm/mysql-core';
+import type { TopLevelCondition } from 'json-rules-engine';
+import { test } from 'vitest';
+import { bigintNarrow, jsonSchema, unsignedBigintNarrow } from '~/column.ts';
+import { CONSTANTS } from '~/constants.ts';
+import { createInsertSchema, createSelectSchema, createUpdateSchema } from '../src';
+import { Expect, expectSchemaShape } from './utils.ts';
+
+const intSchema = type.keywords.number.integer.atLeast(CONSTANTS.INT32_MIN).atMost(CONSTANTS.INT32_MAX);
+const serialNumberModeSchema = type.keywords.number.integer.atLeast(0).atMost(Number.MAX_SAFE_INTEGER);
+const textSchema = type.string.atMostLength(CONSTANTS.INT16_UNSIGNED_MAX);
+
+test('table - select', (t) => {
+	const table = mysqlTable('test', {
+		id: serial().primaryKey(),
+		name: text().notNull(),
+	});
+
+	const result = createSelectSchema(table);
+	const expected = type({ id: serialNumberModeSchema, name: textSchema });
+	expectSchemaShape(t, expected).from(result);
```

Comment:
This is a good pattern- I have tons of tests like this it in ArkType!

However, you will get more useful errors by just doing something like `expect(t.json).deepEqual(expected.json)` without a util.

You can also use `expect(t.expression).deepEqual(expected.expression)` for an even more concise representation of the type that looks like TS, but it won't include metadata like custom descriptions so if you're testing that be sure to use `.json`.